drivers/mm/mm_drv_intel_adsp_mtl_tlb.c

```diff
@@ -702,8 +703,8 @@ static void adsp_mm_save_context(void *storage_buffer)
 	location += sizeof(uint32_t);
 
 	z_xtensa_cache_flush(
-		storage_buffer,
-		(uint32_t)location - (uint32_t)storage_buffer);
+		(void *)storage_buffer,
+		(uint32_t)(void *)location - (uint32_t)(void *)storage_buffer);
```

Comment:
I think this one deserves a sparse-related comment, otherwise it's context-free and very mysterious. I mean _much more_ puzzling than the sparse usual :-)

Wow, three `(void *)` casts in a single function call... Guinness world record? :-) More seriously, how about something a bit lighter like this:

```suggestion
        (uint32_t) ( location - (uint8_t *)storage_buffer  );
```

Reply:
this is really weird and looks wrong. This shouldn't be needed at all. Are you sure sparse is complaining about this? I don't see any such complains in https://github.com/thesofproject/sof/actions/runs/3837426161/jobs/6532722265#logs

Reply:
Weird... my local sparse complains about non-size-preserving casts. I am using version 0.6.4.

Reply:
@dcpleung our current, very top sparse priority is to address all "coherent" sparse warnings. We don't fail on other sparse warnings yet. I suggest you leave them aside for now.

Reply:
@dcpleung ah, that's a different warning, sure. As @marc-hb mentioned, we don't handle them yet. It's certainly good if you want to fix them too for Zephyr, but maybe at least in a separate commit. As for this one - do you run sparse with a 64-bit compiler? On Xtensa pointers are 32 bit. But if or when you decide to fix this, I'd cast pointers to `uintptr_t` instead of `uint32_t`, that should guarantee a proper size integer.